docs/DeveloperGuide.md

```diff
@@ -1,29 +1,342 @@
+
+### 3.4. Command Component
+
+The `Command` component is represented by the `command` package. The `command` package contains all the available user commands supported by the application. These commands are utilised by the user to interact with the expenditure types and the expenditure list. 
+
+The `AcademicExpenditureCommand`, `AccommodationExpenditureCommand`, `BorrowExpenditureCommand`, `EntertainmentExpenditureCommand`, `FoodExpenditureCommand`, `LendExpenditureCommand`, `OtherExpenditureCommand`, `TransportExpenditureCommand`, `TuitionExpenditureCommand` commands contain the operations pertaining to adding a new expenditure into the list of expenditures.
+
+The `CheckBudgetCommand` class class contains the operations pertaining to comparing the total expenditure amount with a budget set by the user. The budget is set with the `SetBudgetCommand`.
+
+The `DeleteCommand` class contains the operations pertaining to deleting a specific expenditure from the list of expenditures.
+
+The `DuplicateCommand` class contains the operations pertaining to duplicating a specific expenditure from the list of expenditures.
+
+The `EditCommand` class contains the operations pertaining to editing a expenditure from the list of expenditures.
+
+The `ExitCommand` class contains the operation that safely closes the application.
+
+The `FindCommand`class contains the operations pertaining to searching the list of expenditures for expenditures that match the keyword entered by the user.
+
+The `HelpCommand` class contains the operation pertaining to providing the user a user interface to the instructions on the use of the application.
+
+The `InvalidCommand` class is instantiated when an unrecognised command is entered by the user.
+
+The `SetBudgetCommand` class contains the operations in setting an amount of money users would like to budget.
+
+The `SortCommand` class contains the operations pertaining to sorting the list of expenditures by amount or by date.
+
+The `ViewDateExpenditureCommand` and `ViewTypeExpenditureCommand` classes contain the operations pertaining to displaying a filtered expenditure list by the expenditure date and type respectively.
+
+Below shows the UML diagram representing the `command` package.
+<p align="center">
+    <img src="team/images/umlCommandClassDiagram.png" width="100%">
+    <br/>
+    <i>Figure 4: UML diagram for the command package</i>
+</p>
+
+A more detailed coverage is explored in [Command List](#4-command-list).
+
+### 3.5. Storage
+The class `TxtFileStatus` and `ExpenditureList` are involved in storing the expenditure list.
+After every user input is completed, saveExpenditureList is called, and the text file will be
+updated with all the current expenditures in the expenditure array list.
+
+<p align="center">
+    <img src="team/images/saveList.png">
+    <br/>
+    <i>Figure 5: Sequence diagram for TxtFileStatus</i>
+</p>
```

Comment:
The activation bar of a method cannot start before the method call arrives and a method cannot remain active after the method has returned. Do take note of this notation error.